src/main/kotlin/com/example/flashcardbackend/deck/DeckRepository.kt

```diff
@@ -0,0 +1,88 @@
+package com.example.flashcardbackend.deck
+
+import com.example.flashcardbackend.flashcard.FlashcardListItem
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.dao.DataAccessException
+import org.springframework.jdbc.core.JdbcTemplate
+import org.springframework.jdbc.core.ResultSetExtractor
+import org.springframework.stereotype.Repository
+import org.springframework.transaction.annotation.Transactional
+import java.sql.ResultSet
+import java.sql.SQLException
+
+@Transactional
+@Repository
+class DeckRepository(@Autowired val jdbcTemplate: JdbcTemplate) {
+
+    fun findAll(): List<DeckListItem> {
+        return jdbcTemplate.query(
+            "SELECT * FROM deck",
+        ) { resultSet, rowNum ->
+            println("Found $rowNum decks")
+            DeckListItem(
+                resultSet.getInt("id"),
+                resultSet.getInt("group_id"),
+                resultSet.getString("deck_name"),
+                resultSet.getString("description"),
+            )
+        }
+    }
+
+    fun insert(deck: DeckCreate): Int {
+        return jdbcTemplate.update(
+            "INSERT INTO deck (deck_name, description, group_id) VALUES (?,?,?)",
+            deck.name,
+            deck.description,
+            deck.deckGroupId,
+        )
+    }
+
+    fun update(deck: DeckUpdate): Int {
+        return jdbcTemplate.update(
+            "UPDATE deck SET deck_name = ?, description = ?, group_id = ? WHERE id = ?",
+            deck.name,
+            deck.description,
+            deck.deckGroupId,
+            deck.id,
+        )
+    }
+
+    fun deleteById(id: Int): Int {
+        return jdbcTemplate.update("DELETE deck WHERE id = ?", id)
+    }
+
+    fun findById(id: Int): Deck? {
+        val sql = """
+            SELECT deck.*, flashcard.id AS flashcard_id, flashcard.deck_id 
+            FROM deck, flashcard 
+            WHERE deck.id = flashcard.deck_id AND flashcard.id = ?
+        """.trimIndent()
+        return jdbcTemplate.query(sql, DeckResultSetExtractor(), id)
+    }
+}
+
+class DeckResultSetExtractor : ResultSetExtractor<Deck?> {
```

Comment:
I just remembered that here is a library called SimpleFlatMapper (https://simpleflatmapper.org/) that should help you get rid of this result set extractor. It's essentially a mapper library that helps you implement read-only one-to-many relationships. I have used it in the past, so, if you run into any problems, let me know. Also, you can fix this in a separate PR.

Reply:
Thank you! I will look into SimpleFlatMapper!

Reply:
Yeah, I think I will fix it in a separate PR